pkg/contract/contract.go

```diff
@@ -1,225 +1,116 @@
 	)
 
-	if validFromStr, ok = params["valid_from"]; !ok {
-		return false, fmt.Errorf("%w: value for [valid_from] is missing", ErrInvalidContractText)
+	if validFromStr, ok = sc.Params["valid_from"]; !ok {
+		return fmt.Errorf("%w: value for [valid_from] is missing", ErrInvalidContractText)
 	}
 
-	validFrom, err = parseTime(validFromStr, c.Language)
+	validFrom, err = parseTime(validFromStr, sc.Template.Language)
 	if err != nil {
-		return false, fmt.Errorf("%w: unable to parse [valid_from]: %s", ErrInvalidContractText, err)
+		return fmt.Errorf("%w: unable to parse [valid_from]: %s", ErrInvalidContractText, err)
 
 	}
 
-	if validToStr, ok = params["valid_to"]; !ok {
-		return false, fmt.Errorf("%w: value for [valid_to] is missing", ErrInvalidContractText)
+	if validToStr, ok = sc.Params["valid_to"]; !ok {
+		return fmt.Errorf("%w: value for [valid_to] is missing", ErrInvalidContractText)
 	}
 
-	validTo, err = parseTime(validToStr, c.Language)
+	validTo, err = parseTime(validToStr, sc.Template.Language)
 	if err != nil {
-		return false, fmt.Errorf("%w: unable to parse [valid_to]: %s", ErrInvalidContractText, err)
+		return fmt.Errorf("%w: unable to parse [valid_to]: %s", ErrInvalidContractText, err)
 	}
 
 	// All parsed, check time range
 	if validFrom.After(*validTo) {
-		return false, fmt.Errorf("%w: [valid_from] must be after [valid_to]", ErrInvalidContractText)
+		return fmt.Errorf("%w: [valid_from] must be after [valid_to]", ErrInvalidContractText)
 	}
 
 	amsterdamLocation, _ := time.LoadLocation("Europe/Amsterdam")
 	now := NowFunc()
 	logrus.Debugf("checking timeframe: now %v, validFrom: %v, validTo: %v", now, *validFrom, *validTo)
 
 	if now.In(amsterdamLocation).Before(*validFrom) {
-		logrus.Infof("contract is not yet valid. now: %s, validFrom: %s", now, validFrom)
-		return false, nil
+		return fmt.Errorf("contract is not yet valid. now: %s, validFrom: %s", now, validFrom)
 
 	}
 	if now.In(amsterdamLocation).After(*validTo) {
-		logrus.Info("contract is expired")
-		return false, nil
+		return fmt.Errorf("contract is expired since: %s", validTo)
 	}
 
-	return true, nil
+	return nil
 }
 
-type Matrix map[Language]map[Type]map[Version]*Template
-
-// ErrUnknownContractFormat is returned when the contract format is unknown
-var ErrUnknownContractFormat = errors.New("unknown contract format")
-
-// ErrInvalidContractFormat indicates tha a contract format is unknown.
-var ErrInvalidContractFormat = errors.New("unknown contract type")
-
-// ErrContractNotFound is used when a certain combination of type, language and version cannot resolve to a contract
-var ErrContractNotFound = errors.New("contract not found")
-
-// ErrInvalidContractText is used when contract texts cannot be parsed or contain invalid values
-var ErrInvalidContractText = errors.New("invalid contract text")
+func parseTime(timeStr string, _ Language) (*time.Time, error) {
+	amsterdamLocation, _ := time.LoadLocation("Europe/Amsterdam")
```

Comment:
const. Also I don't think it shouldn't be named 'amsterdamLocation' but something like 'ContractIssuerTimezone' or something?

Reply:
I think we could add a timezone to the nuts config. Os.timesone sounds like it might break when services are hosted outside of the country of use.